### PR TITLE
[SYCL][ESIMD] Fix reference generation for fp types

### PR DESCRIPTION
**Pull request by yuriykoch**
Reorder branches because any floating point type is signed

Signed-off-by: Kochetkov, Yuriy <yuriyx.kochetkov@intel.com>

### CODE REVIEW

**Comment by yuriykoch** [comment]
@sndmitriev Would you mind to take a look on this small hotfix?

**Review by sndmitriev — APPROVED** [review]
Looks good.

**Comment by yuriykoch** [comment]
Looks like the latest `pre-ci-windows` failed due to the https://github.com/intel/llvm/issues/8816 issue

**Comment by yuriykoch** [comment]
@sndmitriev I'm not authorized to merge this pull request. Would you mind to merge it?

**Comment by sndmitriev** [comment]
I am not authorized as well. @vladimirlaz, can you please help with it?

File: SYCL/ESIMD/api/functional/value.hpp [hunk 1]
```diff
@@ -133,7 +133,24 @@ template <typename DataT, int NumElems> std::vector<DataT> generate_ref_data() {
...
 
-  if constexpr (std::is_signed_v<DataT>) {
+  if constexpr (type_traits::is_sycl_floating_point_v<DataT>) {
+    static const DataT nan = value<DataT>::nan();
+    static const DataT inf = value<DataT>::inf();
+
+    ref_data.reserve((NumElems > 1) ? NumElems : 6);
+
+    // We are using the `double` literals to avoid precision loss for case of
+    // the `double` DataT on unexact values like 0.1
```
Comment (v-klochkov) [inline SYCL/ESIMD/api/functional/value.hpp:143]:
```suggestion
    // the `double` DataT on inexact values like 0.1
```

**Review by v-klochkov — APPROVED** [review]

**Comment by v-klochkov** [comment]
The failed test vector_relational.cpp was disabled about 1 hour ago here: https://github.com/intel/llvm-test-suite/pull/625

**Pull request merged**